collate.py

```diff
@@ -12,43 +22,193 @@ def match_pubs_titleauthor():
+                if score1 > 0.95 and score2 > 0.95:
+                    return True
+                else:
+                    return False
+
+            else:
+                author2_fullname = author2_fname + ' ' + author2_lname
+
+                score1 = SM(None, author1.lower(), author2_fullname.lower()).ratio()
+
+                score2 = SM(None, author1[-len(author2_lname):].lower(), author2_lname.lower()).ratio()
+
+                if score1 > 0.85 or (score1 > 0.65 and score2 > 0.90):
+                    return True
+                else:
+                    return False
+
+        primary_award = []
+        secondary_award = []
+        main_4DN_author_in_pub = 'none'
+        all_4DN_authors_in_pub = []
+        isthere_primary_award = False
+        isthere_4DN_member = False
+        warn1 = False
+        warn2 = False
+
+        wdir = os.getcwd()
+        with open(wdir + "/Pis_and_awards.json", 'r') as f:
+            pi_and_awards_portal = json.load(f)
+
+        # if publication['id'] == "103614":
+        #     import pdb; pdb.set_trace()
+
+        authors_in_pub = publication.get('authors')
+        awards_in_pub = publication.get('awards', None)
+
+        for last_author in reversed(authors_in_pub):
+            for pi in pi_and_awards_portal.values():
+                secondary_award = []
+                fname_pi = pi['first_name']
+                lname_pi = pi['last_name']
+                awards_pi = pi['awards']
+                if is_same_author(last_author, fname_pi, lname_pi):
+                    isthere_4DN_member = True
+                    full_name_pi = fname_pi + ' ' + lname_pi
+                    all_4DN_authors_in_pub.append(full_name_pi)
+                    if awards_in_pub == None or awards_in_pub[0] == "NONE":
+                        awards_in_pub = ['CA200147' if 'TCPA' in award else award[4:-3] for award in awards_pi]
+                    for award_in_pub in awards_in_pub:
+                        for award_of_pi in awards_pi:
+                            select_award = None
+                            if award_in_pub == award_of_pi[4:-3]:
+                                select_award = award_in_pub
+                            elif award_in_pub == 'CA200147' and 'TCPA' in award_of_pi:
+                                select_award = award_of_pi
+                                break
+                            else:
+                                continue
+                        if len(primary_award) == 0 and select_award is not None:
+                            primary_award.append(select_award)
+                            isthere_primary_award = True
+                            main_4DN_author_in_pub = full_name_pi
+                        else:
+                            secondary_award.append(award_in_pub)
+                    break
+                else:
+                    continue
+
+            if isthere_primary_award:
```

Comment:
Is this correct? Don't we want to keep looking for secondary awards?

Reply:
Once we find a primary award. All the other awards become secondary automatically and they are handled before getting to that if statement.